File: Projects/server/Contracts/Responses/TelemetryDataDto.cs
namespace server.Contracts.Responses
{
    public class TelemetryDataDto
    {
        public ulong RecId { get; set; }
        public DateTime RecDatetime { get; set; }
        public byte ServiceId { get; set; }
        public ushort PID { get; set; }
        public string EcuIdBase64 { get; set; } = null!;
        public byte ResponseDlc { get; set; }
        public string? ResponseBase64 { get; set; }
        public ulong TripId { get; set; }
    }
}

File: Projects/DataProviderApp/app/src/main/java/com/example/dataproviderapp/dto/responses/TelemetryDataDto.kt
package com.example.dataproviderapp.dto.responses

import java.time.LocalDateTime

data class TelemetryDataDto(
    val recId: ULong,
    val recDatetime: LocalDateTime,
    val serviceId: UByte,
    val pid: UShort,
    val ecuIdBase64: String,
    val responseDlc: UByte,
    val responseBase64: String?,
    val tripId: ULong
)
